assets/javascripts/discourse/templates/components/chat-channel-about-view.hbs

```diff
@@ -71,21 +71,12 @@
 {{/if}}
 
 <div class="chat-form__section">
-  {{#if channel.following}}
-    {{d-button
-      action=(action "onLeaveChannel")
-      label="chat.channel_settings.leave_channel"
-      class="chat-form__btn leave-channel-btn btn-flat"
-      icon="sign-out-alt"
-      disabled=isLoading
-    }}
-  {{else}}
-    {{d-button
-      action=(action "onJoinChannel")
-      label="chat.channel_settings.join_channel"
-      class="chat-form__btn join-channel-btn btn-flat"
-      icon="sign-in-alt"
-      disabled=isLoading
-    }}
-  {{/if}}
+  <ChatToggleMembershipBtn
```

Comment:
So for the public API I would try to follow this: mandatory -> top level, optional -> some options hashs

```
<ToggleChannelMembershipButton
  @channel={{...}}
  @onToggle={{...}}
  @options={{hash
    joinIcon="bell"
    leaveClass="foo"
  }}
/>
```

Note:
- I changed the name, in the future I will drop the chat from many names, and I think it makes more sense to use channel here anyways
- s/Btn/Button, for parity with d-button
- onToggle as mentioned sooner
- class and not classes, if you look at html we say class even if you can pass multiple values, this is the class value

Reply:
All great points, thanks